datafusion/src/scalar.rs

```diff
@@ -324,8 +324,8 @@ impl ScalarValue {
     ///
     /// assert_eq!(&array, &expected);
     /// ```
-    pub fn iter_to_array<'a>(
-        scalars: impl IntoIterator<Item = &'a ScalarValue>,
+    pub fn iter_to_array(
+        scalars: impl IntoIterator<Item = ScalarValue>,
```

Comment:
Yeah, this is unfortunate -- I was trying to hard to avoid the need for owned `ScalarValues` -- but I think since `SclarValues` effectively own the underlying storage, if the source data is in some other form, you end up having to create one anyways.

But I think this change is for the better; 👍